bundle/src/main/java/com/adobe/acs/commons/mcp/AdministratorsOnlyProcessDefinitionFactory.java

```diff
@@ -20,36 +20,17 @@
 package com.adobe.acs.commons.mcp;
 
 import aQute.bnd.annotation.ConsumerType;
-import org.apache.jackrabbit.api.security.user.User;
-
-import javax.jcr.RepositoryException;
-import java.util.Spliterator;
-import java.util.Spliterators;
-import java.util.stream.StreamSupport;
 
 /**
  * ProcessDefinitionFactory which limits availability of a process to the admin users and members of the administrators group.
  * @param <P> Process definition generated by this factory
  */
 @ConsumerType
-public abstract class AdministratorsOnlyProcessDefinitionFactory<P extends ProcessDefinition> extends ProcessDefinitionFactory<P> {
+public abstract class AdministratorsOnlyProcessDefinitionFactory<P extends ProcessDefinition> extends AuthorizedGroupProcessDefinitionFactory<P> {
 
     @Override
-    @SuppressWarnings("squid:S1141")
-    public boolean isAllowed(User user) {
-        try {
-            return user.isAdmin()
-                    || StreamSupport.stream(Spliterators.spliteratorUnknownSize(user.memberOf(), Spliterator.ORDERED),false)
-                    .anyMatch(g -> {
-                        try {
-                            return g.getID().equals("administrators");
-                        } catch (RepositoryException e) {
-                            return false;
-                        }
-                });
-        } catch (RepositoryException e) {
-            return false;
-        }
+    public String[] getAuthorizedGroups() {
```

Comment:
this method should probably be final. Also it is missing the @Override annotation AFAICT

Reply:
It has the annotation but making it final makes sense.